### PR TITLE
whisper: init at 2.0.1

### PR DESCRIPTION
**Pull request by jbedo**
###### Motivation for this change

Whisper init.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/whisper/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchFromGitHub, fetchurl }:
+
+stdenv.mkDerivation {
+  pname = "whisper";
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner = "refresh-bio";
+    repo = "Whisper";
+    rev = "v2.0";
```
Comment (jollheef) [inline pkgs/applications/science/biology/whisper/default.nix:10]:
```suggestion
    rev = "v${version}";
```

https://nixos.wiki/wiki/Nix_Expression_Language#rec_statement

File: pkgs/applications/science/biology/whisper/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "100p2pqhli123wnqkxrvjcwnmlvcp1rk63whh121jannnw81rh2m";
+  };
+
+  patches = [ (fetchurl {
+    url = "https://github.com/refresh-bio/Whisper/commit/9ad77e7de68d91d9427cfbe6211e83cea89206ab.patch";
```
Comment (jollheef) [inline pkgs/applications/science/biology/whisper/default.nix:15]: Can we ask repository owners to bump a minor version instead?

**Review by jollheef — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/whisper/default.nix (hunk 2 above)
Comment (jbedo) [reply pkgs/applications/science/biology/whisper/default.nix:15]: Can always ask! I'll open an issue.

Comment (jbedo) [reply pkgs/applications/science/biology/whisper/default.nix]: Expression updated and tested with new release.

File: pkgs/applications/science/biology/whisper/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchFromGitHub, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "whisper";
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "refresh-bio";
+    repo = "Whisper";
```
Comment (jollheef) [inline pkgs/applications/science/biology/whisper/default.nix:9]: Here you can also use 
```suggestion
    repo = pname;
```

**Review by jollheef — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/biology/whisper/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  '';
+
+  buildPhase = "make -j $NIX_BUILD_CORES";
+
+  meta = with stdenv.lib; {
+    description = "Short read sequence mapper";
+    license = licenses.gpl3;
+    homepage = "https://github.com/refresh-bio/whisper";
+    maintainers = with maintainers; [ jbedo ];
+    platforms = platforms.linux;
```
[the remark this answers is not in the document]
Comment (jbedo) [reply pkgs/applications/science/biology/whisper/default.nix]: The documentation only mentioned windows and linux, but I think it should work on macos. I don't have macos to try, so I've updated platforms to all and we'll see if CI builds.

**Comment by jbedo** [comment]
Updated for x86_64 only.

**Review by jollheef — APPROVED** [review]
LGTM

**Comment by balsoft** [comment]
Why is `buildPhase` set explicitly? If there is a reason, please respect `enableParallelBuilding` attribute.

**Comment by jbedo** [comment]
Removed, was unnecessary.

File: pkgs/applications/science/biology/whisper/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  preConfigure = ''
+    cd src
+
+    # disable default static linking
+    sed -i 's/ -static / /' makefile
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp whisper whisper-index $out/bin
```
Comment (jtojnar) [inline pkgs/applications/science/biology/whisper/default.nix:23]:
```suggestion
    runHook preInstall

    install -Dt $out/bin whisper whisper-index

    runHook postInstall
```

Comment (jbedo) [reply pkgs/applications/science/biology/whisper/default.nix]: Done.

**Pull request merged**